mentorship ios/Views/Registration/Login.swift

```diff
@@ -30,14 +32,29 @@ struct Login: View {
 
             //login button
             Button("Login") {
-                self.loginViewModel.login()
+                // set inActivity to true (shows activity indicator)
```

Comment:
```
//
//  LoginView.swift
//  Created on 01/06/20.
//  Created for AnitaB.org Mentorship-iOS 
//

import SwiftUI

struct Login: View {
    var loginService: LoginService?
    @State private var showSignUpPage: Bool = false
    @State private var inActivity = false
    @ObservedObject var loginViewModel = LoginViewModel()

    var body: some View {
        VStack(spacing: DesignConstants.Form.Spacing.bigSpacing) {
            //top image of mentorship logo
            Image(ImageNameConstants.mentorshipLogoImageName)
                .resizable()
                .scaledToFit()

            //username and password text fields
            VStack(spacing: DesignConstants.Form.Spacing.smallSpacing) {
                TextField("Username/Email", text: $loginViewModel.username)
                    .textFieldStyle(RoundFilledTextFieldStyle())
                    .autocapitalization(.none)
                    .keyboardType(.emailAddress)

                SecureField("Password", text: $loginViewModel.password)
                    .textFieldStyle(RoundFilledTextFieldStyle())
            }

            //login button
            Button("Login") {
                // set inActivity to true (shows activity indicator)
                self.inActivity = true
                let loginData = LoginModel.LoginUploadData( username: self.loginViewModel.username,password: self.loginViewModel.password)
                self.loginService?.login(loginData: loginData) { response in
                    self.inActivity = false
                    self.loginViewModel.message = response.message
                }
            }
            .buttonStyle(BigBoldButtonStyle(disabled: loginViewModel.loginDisabled))
            .disabled(loginViewModel.loginDisabled)

            //activity indicator or show user message text
            if inActivity {
                ActivityIndicator(isAnimating: $inActivity)
            } else {
                Text(self.loginViewModel.message ?? "")
                    .modifier(ErrorText())
            }

            //text and sign up button
            VStack(spacing: DesignConstants.Form.Spacing.minimalSpacing) {
                Text(LocalizableStringConstants.noAccountText)

                Button.init(action: { self.showSignUpPage.toggle() }) {
                    Text("Signup")
                        .foregroundColor(DesignConstants.Colors.defaultIndigoColor)
                }.sheet(isPresented: $showSignUpPage) {
                    SignUp(isPresented: self.$showSignUpPage)
                }
            }

            //spacer to push content to top
            Spacer()
        }
        .modifier(AllPadding())
    }
}

struct LoginView_Previews: PreviewProvider {
    static var previews: some View {
        Login()
    }
}
```

Reply:
Why are we preferring an optional service instead of using default parameters? I think, using default parameters is better and will make updates easier. Also, we can still explicitly inject a service. For this reason, I haven't modified the content view.
Let me know if I'm missing something

Reply:
as we discussed I'm ok either way.